app/lib/azureDataService.js

```diff
@@ -0,0 +1,50 @@
+const azureService = require('./azureService');
+const config = require('./config');
+const fsHelper = require('./fsHelper');
+const log = require('./logger');
+const moment = require('moment');
+const utils = require('./utils');
+const createFileVersionFilter = require('./createFileVersionFilter');
+
+const outputFile = `${config.outputDir}/${config.outputFile}.json`;
+const summaryFile = `${config.outputDir}/summary.json`;
+
+async function getLatestDataBlob(version) {
+  return azureService.getLatestBlob(createFileVersionFilter(config.outputFile, version));
+}
+
+async function getLatestData(version) {
+  const lastScan = await getLatestDataBlob(version);
+  if (lastScan) {
+    log.info(`Latest data file '${lastScan.name}' identified`);
+    await azureService.downloadFromAzure(outputFile, lastScan.name);
+    log.info(`Latest data file '${lastScan.name}' downloaded`);
+    const data = fsHelper.loadJsonSync(config.outputFile);
+    const date = moment(lastScan.lastModified);
```

Comment:
I don't think `lastModified` should be used. Instead the date embedded within the file is 'better'. The reasons are; it is more consistent, it doesn't change should the file get edited after the date it was originally uploaded. It also doesn't matter if the run starts on one day and finishes on the next day as the date of the file will be for the date of the run (actually this isn't true as the datestamp is for the day it is uploaded - that could do with changing).

I've made the change in https://github.com/nhsuk/pharmacy-data-etl/compare/feature/tweaks?expand=1#diff-e62e3b8bb63f34fc49bfd6452ad8a57bR46

Reply:
I've changed the azure data service to get the date from the name, and used the ETL start date as the filestamp in the ETL